deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -152,6 +153,11 @@ class FileSystemDeploymentService implements DeploymentScanner, NotificationHand
     @SuppressWarnings("deprecation")
     private final DeploymentTransformer deploymentTransformer;
 
+    private ScanContext scanContext;
+
+    public Map<String,Boolean> getScanContextRegisteredDeployments() {
+        return Collections.unmodifiableMap(scanContext.registeredDeployments);
```

Comment:
Per my comment above about the new scanContext field, please wrap this in acquireScanLock() try { ... } finally { releaseScanLock(); }

an 'assert scanContext != null' would also be good.

Doing all this doesn't really matter for your test but it reduces the risk of future misuse of this method by having the test code follow the concurrency rules that we create for the production code.